ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Keystore/Types.hs

```diff
@@ -0,0 +1,92 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Ariadne.Wallet.Cardano.Kernel.Keystore.Types
+       ( StorageDecodingError (..)
+       , UserSecret (..)
+       , Version (..)
+       , eskToKey
+       , getUserSecretKeys
+       , usFilePath
+       , usWallets
+       , usVersion
+       ) where
+
+import Control.Lens.TH (makeLenses)
+import Data.Aeson
+  (FromJSON, ToJSON, Value(..), toJSON, parseJSON, withText, withObject, (.:), object, (.=))
+import Data.Default
+import Data.Map (Map)
+import Formatting (bprint, build)
+
+import Pos.Binary.Class (serialize', decodeFull')
+import Pos.Core (AddressHash, addressHash)
+import Pos.Crypto (EncryptedSecretKey(..), PublicKey, encToPublic)
+
+import qualified Data.Text.Buildable
+import qualified Data.Map as Map (empty, fromList, elems)
+import qualified Serokell.Util.Base16 as S (encode, decode)
+
+newtype Version= Version { getVersion :: Int }
+  deriving (Show, Eq)
+
+-- | UserSecret data containing wallet keys that gets saved to file
+data UserSecret
+  = UserSecret
+  { _usWallets  :: Map (AddressHash PublicKey) EncryptedSecretKey
+  , _usVersion  :: !Version
+  , _usFilePath :: !FilePath
+  } deriving (Show)
+
+makeLenses ''UserSecret
+
+instance Default UserSecret where
```

Comment:
Since this instance is _always_ used like so
```haskell
def & usFilePath .~ fp
```
perhaps it would be better to provide a separate function, like `makeUserSecret :: FilePath -> UserSecret` or something like that?